modules/tensor_mechanics/src/materials/ComputeElasticityBeam.C

```diff
@@ -0,0 +1,60 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "ComputeElasticityBeam.h"
+#include "Function.h"
+
+template <>
+InputParameters
+validParams<ComputeElasticityBeam>()
+{
+  InputParameters params = validParams<Material>();
+  params.addParam<FunctionName>(
+      "elasticity_prefactor",
+      "Optional function to use as a scalar prefactor on the elasticity vector for the beam.");
+  params.addCoupledVar("youngs_modulus", "Variable containing Youngs modulus");
+  params.addCoupledVar("shear_modulus", "Variable contiaining shear modulus");
+  params.addCoupledVar("shear_coefficient", "Reduction factor for the shear modulus");
```

Comment:
I'd prefer "scale factor" over "reduction factor" in this description.  Also, I think I'd change the descriptions of those coupled variables to say something like "Young's modulus of the material. Can be supplied as either a number or a variable name." ( so it's less confusing to the typical user who will probably just supply a number rather than a variable).